### PR TITLE
Make Element requirement in TensorArrayProtocol.init a conformance.

### PR DESCRIPTION
**Pull request by bgogul**
Related to https://bugs.swift.org/browse/TF-542

### CODE REVIEW

**Comment by rxwei** [comment]
Do you ever need to call `init(_handles:)` on an array of type-erased `_AnyTensorHandle`s? If so, this change would make that impossible.

File: Sources/TensorFlow/Core/TensorGroup.swift [hunk 1]
```diff
@@ -35,7 +35,7 @@ public protocol TensorArrayProtocol {
     var _tensorHandles: [_AnyTensorHandle] { get }
 
     init(_owning tensorHandles: UnsafePointer<CTensorHandle>?, count: Int)
-    init<C: RandomAccessCollection>(_handles: C) where C.Element == _AnyTensorHandle
+    init<C: RandomAccessCollection>(_handles: C) where C.Element : _AnyTensorHandle
```
Comment (rxwei) [inline Sources/TensorFlow/Core/TensorGroup.swift]: The Google Swift Style Guide recommends no whitespaces before colons in a conformance constraint.

```suggestion
    init<C: RandomAccessCollection>(_handles: C) where C.Element: _AnyTensorHandle
```

**Comment by bgogul** [comment]
> Do you ever need to call `init(_handles:)` on an array of type-erased `_AnyTensorHandle`s? If so, this change would make that impossible.

There is one use case in the test. That is why I made the following change: 
```
+copy(of: TensorHandle<T>) -> TFETensorHandle
-copy(of: TensorHandle<T>) -> _AnyTensorHandle
```

Without this change, I was not able to initialize a TensorGroup with an array of specific type, e.g.,  `[TFETensorHandle] or `[LazyTensor]`. Is there a better way to achieve this?

**Comment by rxwei** [comment]
A `[TFETensorHandle]` should be directly convertible to a `[_AnyTensorHandle]` because it's a subtype.

**Comment by bgogul** [comment]
> A `[TFETensorHandle]` should be directly convertible to a `[_AnyTensorHandle]` because it's a subtype.

Ah, that is right. The problem that I faced was with using  an `ArraySlice<TFETensorHandle>` as argument to `init(_handles:_)`.

File: Sources/TensorFlow/Core/TensorGroup.swift [hunk 2]
```diff
@@ -290,7 +290,8 @@ extension Array: TensorArrayProtocol where Element: TensorGroup {
 
     public init<C: RandomAccessCollection>(
         _handles: C
-    ) where C.Element == _AnyTensorHandle {
+    ) where C.Element: _AnyTensorHandle {
+
```
Comment (rxwei) [inline Sources/TensorFlow/Core/TensorGroup.swift]:
```suggestion
```

**Comment by rxwei** [comment]
I see. Yeah, variance only works with a few stdlib types in Swift such as `Array`.

File: Sources/TensorFlow/Core/TensorGroup.swift (hunk 2 above)
Comment (bgogul) [reply Sources/TensorFlow/Core/TensorGroup.swift]: Done.

**Pull request merged**